### PR TITLE
Release build workflow updates

### LINKED ISSUE #1209: Fix release workflow builds
The installation archive release builds are failing. Two fixes are needed:
1. Set BOOST_ROOT to the location of Boost 1.72.0
2. Update the Docker image used for Linux builds

### PR DESCRIPTION
**Pull request by nightlark**
### Summary
If merged this pull request will update the Linux Docker container and Boost versions, and fix the Windows builds (VM image removed Boost from `PATH`). This fixes the failing nightly runs for the release build workflow.

Closes #1209

### Proposed changes
- Update Linux release build to use Boost 1.72.0
- Update Linux release build Docker container to use the latest base image (contains devtoolset-8 and git by default)
- Fix Windows builds by setting BOOST_ROOT to Boost 1.72.0 location

### CODE REVIEW

File: .github/actions/linux-release-builder/Dockerfile [hunk 1]
```diff
@@ -2,9 +2,8 @@ FROM phusion/holy-build-box-64:latest
 
 # Install a copy of git and the Boost headers
 # curl -L follows redirects and -k ignores SSL certificate warnings
-RUN yum install -y rh-git29 devtoolset-8 \
-    && curl -Lk -o boost.tar.gz https://sourceforge.net/projects/boost/files/boost/1.70.0/boost_1_70_0.tar.gz/download \
-    && tar --strip-components=1 -xf boost.tar.gz boost_1_70_0/boost \
+RUN curl -Lk -o boost.tar.gz https://sourceforge.net/projects/boost/files/boost/1.72.0/boost_1_72_0.tar.gz/download \
```
Comment (phlptp) [inline .github/actions/linux-release-builder/Dockerfile:5]: do we not need the yum install lines any more?

Comment (nightlark) [reply .github/actions/linux-release-builder/Dockerfile:5]: The new base image has devtoolset-8 installed and a copy of git.

**Comment by phlptp** [comment]
whats up with the doc failure?  I am assuming nothing to do with this PR?

**Comment by nightlark** [comment]
This only changes files in the `.github` folder, nothing related to docs.

**Comment by phlptp** [comment]
@kdheepak any idea what is going on with the docs in this PR?

**Comment by nightlark** [comment]
I'm guessing this is a flake with the readthedocs service, given that none of the changed files are involved in building docs.

**Comment by nightlark** [comment]
I asked on the RTD gitter, they were having issues earlier (maybe related to the new release they had today).

**Pull request merged**